20_classes_lab/03.py

```diff
@@ -11,3 +11,40 @@
 Implement widget so the following code works
 """
 
+class Widget(object):
+
+    def __init__(self, name):
+        self._name = name
+        self._dependencies = []
+        self._is_built = False
+
+    def add_dependency(self, *widgets):
+        self._dependencies.extend(widgets)
+
+    def build(self):
+        if (self._is_built): return
```

Comment:
Nice! Also you don't need parenthesis here, so this is nicer: `if self._is_built: return`